components/Dialogs/src/MDCDialogPresentationController.m

```diff
@@ -197,9 +197,26 @@ - (CGSize)sizeForChildContentContainer:(id<UIContentContainer>)container
     return CGSizeZero;
   }
 
+  // For pre iOS 11 devices, we are assuming a safeAreaInset of UIEdgeInsetsZero
+  UIEdgeInsets containerSafeAreaInsets = UIEdgeInsetsZero;
+#if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
+  if (@available(iOS 11.0, *)) {
+    containerSafeAreaInsets = self.containerView.safeAreaInsets;
+  }
+#endif
+
+  // Take the larger of the Safe Area insets and the Material specified insets.
+  CGFloat standardWidthInset = MDCDialogEdgeInsets.left + MDCDialogEdgeInsets.right;
+  CGFloat safeAreaWidthInset = containerSafeAreaInsets.left + containerSafeAreaInsets.right;
+  CGFloat widthInset = MAX(standardWidthInset, safeAreaWidthInset);
```

Comment:
Since the safeAreaInsets can be asymmetric I don't believe your centering logic won't work correctly anymore.

Reply:
Good catch.  Split inset code out from sizing code to allow handling this case.